examples/tv-casting-app/android/App/app/src/main/jni/com/chip/casting/DiscoveredNodeData.java

```diff
@@ -216,4 +217,25 @@ public String toString() {
         + ipAddresses
         + '}';
   }
+
+  // autogenerated
+  @Override
+  public int hashCode() {
+    int result = Objects.hash(vendorId, productId, commissioningMode, deviceType);
+    result = 31 * result + Arrays.hashCode(rotatingId);
+    return result;
+  }
+
+  // autogenerated
```

Comment:
what auto-generates this?

It seems somewhat confusing that hash and equality uses 5 values where as the class itself is a container of 16 elements.

Would it make sense to somehow have a hashable/keyable `getKey` or `getId` or similar that returns a container of the 4 relevant items? Otherwise we should heavily document that these are not really using all data within the object.

Reply:
This was generated by AndroidStudio - i should have specified that it is auto generated but i stripped out some elements to it. The main reason being i still wanted it to match the "equals" clause even in the cases where those other elements (like display name) didn't match because the point was that it was "effectively equal" - in this case despite the change in elements such as the broadcast name the DiscoveredNode still represented the same object.

Reply:
But if anything in a struct is different from another struct, they are not 'equal' right?

Put it in another way: in a unit test, if I validate scanner behavior I should be able to rely on equality for testing and this is incomplete. The "auto-generated" comment is misleading then: this was auto-generated and edited on purpose. this needs at least documentation (and if I would be a maintainer of the app I would request a separate class encapsulating only items I actually care about).

Reply:
Thats fair, i changed the implementation to remove it from equals and introduce a new `discoveredNodeHasSameSource` method that takes care of the equality check and makes it more obvious the objects are not exactly equal.